core/src/main/scala/io/github/timwspence/cats/stm/STM.scala

```diff
@@ -194,10 +197,10 @@ object STM {
 
   private[stm] object internal {
 
-    type TLog    = MMap[Long, TLogEntry]
-    type Pending = () => Unit
+    private[stm] type TLog    = MMap[Long, TLogEntry]
```

Comment:
Do these add anything over the declaration on `internal`?

(Hi, I've been doing Scala for 10 years and don't know this.)

Reply:
The type was leaking via the stm run method previously. So I believe so, but it may not.

Reply:
Yeah, this is a bit nasty and surely shouldn't be necessary?

Good spot on the constructor and `run` for `STM` though - `STM` should definitely be opaque.

Sorry if this is a dumb question: why add `final` on `TQueue`, etc? Is it for performance? Or just because there's no real reason why someone should be sub-classing?

Reply:
Final is to specifically prevent anyone from subclassing. People doing that allows people to dynamically circumvent logic, whereas finalizing it means we know the total coherent logic holds in all cases.

As far as `private[stm]` inside the private object is concerned. It may not be necessary, but I also don't see what it hurts. We can remove these and hope that it was seeing that type and unwrapping it because aliases sometimes do that.

Reply:
We bumped the version because Main was on the classpath. So we're in a place where binary incompatible changes are presently allowed without a version change.